retrieval/scrape.go

```diff
@@ -763,6 +767,14 @@ loop:
 			var lset labels.Labels
 			mets := p.Metric(&lset)
 
+			for _, l := range lset {
+				if !utf8.ValidString(l.Value) {
+					err = errInvalidLabelValue
+					sl.l.With("timeseries", string(met)).Debug("Invalid UTF-8 label value")
+					break loop
+				}
+			}
```

Comment:
That's updated.

Also spotted a deviation from the text format that's easy to fix.